packages/workflow/src/workflow.ts

```diff
@@ -832,18 +836,15 @@ export async function executeChild<T extends Workflow>(
  * }
  */
 export function workflowInfo(): WorkflowInfo {
-  if (state.info === undefined) {
-    throw new IllegalStateError('Workflow uninitialized');
-  }
-  return state.info;
+  return getActivator().info;
 }
 
 /**
  * Returns whether or not code is executing in workflow context
  */
 export function inWorkflowContext(): boolean {
   try {
-    workflowInfo();
+    getActivator();
```

Comment:
Seems simple enough to avoid depending on an exception here...

```
  return (globalThis as any).__TEMPORAL__?.activator !== undefined;
```

I can't find any up-to-date performance data on this subject, but I generally assume a performance penalty on throwing/catching exceptions...

Reply:
Hmm.. I could probably add a `maybeGetActivator()` helper.
I'm trying to minimize the use of the untyped global object.

Reply:
I might do this in a later PR, would rather merge for now.